### DISCUSSION TITLE
How to mount certificates (secrets) on Airflow KubernetesExecutor worker pods

### DISCUSSION THREAD

**Discussion by mlnsharma** [apache/airflow#16240] [General] [answered]
I am using Airflow 1.10.15 with KubernetesExecutor with Postgres backend. 

According to http://airflow.apache.org/docs/apache-airflow/1.10.15/configurations-ref.html#kubernetes-secrets, secrets can be mounted as environment variables. 

According to https://airflow.apache.org/docs/apache-airflow-providers-postgres/stable/connections/postgres.html, we need to provide certificate paths in the URL

`export AIRFLOW_CONN_POSTGRES_DEFAULT='postgresql://postgres_user:XXXXXXXXXXXX@1.1.1.1:5432/postgresdb?sslmode=verify-ca&sslcert=/tmp/client-cert.pem&sslkey=/tmp/client-key.pem&sslrootcert=/tmp/server-ca.pem'`

So, how do we load the secrets as files on Worker pods ?

**Comment by jedcunningham** [comment, marked as answer]
Using `pod_template_file` is probably the easiest solution, and sets you up well for upgrading to 2.0+: https://github.com/apache/airflow/blob/5786dcdc392f7a2649f398353a0beebef01c428e/airflow/config_templates/default_airflow.cfg#L825
https://airflow.apache.org/docs/apache-airflow/stable/executor/kubernetes.html#pod-template-file  (Added in 1.10.12)

If that won't work, [pod_mutation_hook](https://airflow.apache.org/docs/apache-airflow/1.10.15/kubernetes.html#pod-mutation-hook) (careful, it applies to _all_ pods, not just workers) or [executor_config](https://github.com/apache/airflow/blob/5786dcdc392f7a2649f398353a0beebef01c428e/airflow/example_dags/example_kubernetes_executor_config.py#L68) (probably least desirable as you'd have to modify every DAG) should both also be options.

**Reply by jedcunningham** [reply]
Forgot to mention, you can generate your pod_template_file with [generate_pod_tempalte](https://airflow.apache.org/docs/apache-airflow/1.10.15/cli-ref.html#generate_pod_template), which should take the tedium out of moving to that approach.

**Reply by mlnsharma** [reply]
Thanks, I found this solution helpful
